### PR TITLE
show error when toggle publish fails

### PR DESCRIPTION
**Pull request by jerminatorhits**
## Description
Why did you write this code?
Goes with https://github.com/thebeetoken/beenest-backend/pull/732
Before, error is only thrown in the console. This pops up an alert when publish fails.

## How to Test
- [ ] Switch Backend branch to https://github.com/thebeetoken/beenest-backend/pull/732
- [ ] Signup using a fake email.
- [ ] Visit 'host/listings' and make a new listing
- [ ] Toggling the publish should fail

Which devices did you test on?
- [ ] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

## REVIEWERS:
Check against these principles:

### High level
Does this code need to be written?
What are the alternatives?
Will this implementation become a support issue?
How much error margin does this solution have?

### Code
* Does the code follow industry standards?
JS: https://github.com/airbnb/javascript
React: https://github.com/airbnb/javascript/tree/master/react
https://github.com/vasanthk/react-bits
Documentation headers: http://usejsdoc.org/index.html

* Is there duplicated code? Can it be refactored into a shared method?
* Is the code consistent with our project?
* Are there unit tests? Do they test the states?
* Is the person refactoring another developer's code? If possible, did the original developer approve?

### Variables/Naming:
* Would the variable type led to future edge cases?
* Are the variable naming clear? Would the value contain something other than what the name describes.

### Security
* Can this be hacked or abused by the user?

## Further Work
Will this have future work?

## Learnings
Did you learn anything here you want to share with the team?

### CODE REVIEW

**Comment by jerminatorhits** [comment]
alert shows `You are not authorized to activate this listing` which could be misleading. Maybe there should be a prompt telling the user to verify and if so, should that text be placed in the front end or have it thrown via backend.

**Comment by jerminatorhits** [comment]
@woeltjen makes sense to me. I'll include it in this PR!

One small possible confusion could be that the unverified user completed all the fields but still cannot publish, but maybe it's implied that they need to verify since the banner is there?
image

File: src/components/routes/Host/HostListingCard/HostListingCard.tsx [hunk 1]
```diff
@@ -79,12 +80,18 @@ const HostListingCard = (props: Props): JSX.Element => {
...
+                      checked={isActive}
+                      disabled={!canPublish || !completedVerification}
+                      onColor={hexColor('correct')}
+                      onChange={() => toggleListing(id).catch((error) => alert(error))}
+                      id={`publish-${id}`} />
+                  );
+                }}
+              </FirebaseConsumer>
             </label>
             {!canPublish && <p className='host-listing-notice'>{INCOMPLETE_LISTING}</p>}
```
Comment (tc) [inline src/components/routes/Host/HostListingCard/HostListingCard.tsx]: maybe we can do ``!completedVerification ^^ "You must verify your email and phone to publish."``

Comment (jerminatorhits) [reply src/components/routes/Host/HostListingCard/HostListingCard.tsx]: 👍

**Pull request merged**